source/features/split-issue-pr-search-results.tsx

```diff
@@ -47,11 +78,13 @@ function init(): void {
 	issueLink.after(prLink);
 
 	const title = select('.codesearch-results h3')!.firstChild!;
-	if (pageSearchQuery.includes('is:pr')) {
+
+	const is = searchQueryIs(searchQuery);
+	if (is === 'pr') {
```

Comment:
Instead of this change, perhaps SearchQuery can handle deduplicating opposite flags (issue/pr for this bugfix; later for open/closed, merged/unmerged, etc)

Reply:
Sure why not 👍
I think `SearchQuery` just looked a bit scary to touch with it's side effects 😄

Reply:
When should the query be cleaned?
Right now it is only cleaned when it is `set`, so I added the deduplicate code there.
_(This allows to just use `add` [here](https://github.com/sindresorhus/refined-github/pull/3837#pullrequestreview-557362968))_

Should it also be cleaned on `get` then we could use the `includes` here.
Alternative would also be to add something linke `last(...searchStrings: string[]): string | null {}`.

Reply:
As for the reloading, I think it should stop accepting a raw Location, perhaps it should be changed to use the `location.search` string, this way we eliminate the side effect. 

I’m conflicted about where this should go, because it should be a .clean()/.deduplicate() method to make it explicit, but also we don’t want to have to manually call it every time. I think they should be deduplicated:

- in the constructor, for existing values
- in set, for any new values 

So maybe you can implement it as a private method and call it from both places

Reply:
Okay also implemented this now, I just did `this.set(this.get())` to clean the initial query string.
This also means it will update the search for all passed *"sources"*.